clang/test/Sema/alias-unused-win.cpp

```diff
@@ -0,0 +1,27 @@
+// RUN: %clang_cc1 -triple %ms_abi_triple -Wunused -x c -verify %s
+// RUN: %clang_cc1 -triple %ms_abi_triple -Wunused -verify=expected,cxx %s
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+static int f(void) { return 42; } // cxx-warning{{unused function 'f'}}
+int g(void) __attribute__((alias("f")));
+
+static int foo [] = { 42, 0xDEAD };
+extern typeof(foo) bar __attribute__((unused, alias("foo")));
+
+static int __attribute__((overloadable)) f0(int x) { return x; } // expected-warning{{unused function 'f0'}}
+static float __attribute__((overloadable)) f0(float x) { return x; } // expected-warning{{unused function 'f0'}}
+int g0(void) __attribute__((alias("?f0@@YAHH@Z")));
```

Comment:
As someone not familiar with microsoft's ABI, is there a tool like `llvm-cxxfilt` that can help me check this mangling from the command line, on my Linux host?

---

I'm guessing this test file is demonstrating the lack of msabi mangling support? And will function as a change detector in the future?

Reply:
> I'm guessing this test file is demonstrating the lack of msabi mangling support? And will function as a change detector in the future?

Yes, a change detector. The test shall be pre-committed once good enough.

llvm-cxxfilt seems Itanium only.... I compiled the file locally and used `llvm-objdump -t a.o` to get the symbol name.

Reply:
FWIW, I use http://demangler.com/ to help check manglings, which isn't quite what you were asking for but might help.

Reply:
got it; please add a TODO that the overload of `f0` that accepts an `int` should not warn (similar to the TODO you added in clang/test/Sema/alias-unused.cpp for the namespace part.  Same for `g1` and the overload of `f3` that accepts no parameters.